[DOC-10324] Added new file for function UNNEST_POS

sarahlwelton opened the pull request:
Added two new example files, unnest-pos.jsonc and unnest-pos.n1ql.

Added file for function, unnestposfun.adoc, to nav.adoc in N1QL lang ref

Added a tip to unnest.adoc to mention UNNEST_POS function.

@sitaramv , I've added you as a reviewer because you were kind enough to help give some context when I asked in the #n1ql channel. Please let me know if I've accurately captured what you shared with me!

sarahlwelton committed:
[DOC-10324] Added new file for function UNNEST_POS

Added two new example files, unnest-pos.jsonc and unnest-pos.n1ql.

Added file for function, unnestposfun.adoc, to nav.adoc in N1QL lang ref

Added a tip to unnest.adoc to mention UNNEST_POS function.

sarahlwelton committed:
[DOC-10324] Correcting langauge in unnestposfun.adoc

simon-dew commented on modules/ROOT/nav.adoc:
```diff
@@ -117,6 +117,7 @@
    **** xref:n1ql:n1ql-language-reference/metafun.adoc[Miscellaneous Utility Functions]
     ***** xref:n1ql:n1ql-language-reference/advisor.adoc[ADVISOR Function]
     ***** xref:n1ql:n1ql-language-reference/curl.adoc[CURL Function]
+    ***** xref:n1ql:n1ql-language-reference/unnestposfun.adoc[UNNEST_POS Function]
```
Don't think this function needs a page by itself? CURL() and ADVISOR() do need a page each, because they are massive, but this is just a pretty straighforward function. Happy to classify this as under "miscellaneous" but I would add it to `metafun.adoc`.

simon-dew commented on modules/n1ql/examples/functions/unnest-pos.n1ql:
```diff
@@ -0,0 +1 @@
+SELECT UNNEST_POS(u) AS upos, u FROM [{"a1":[10,9,4]}] AS d UNNEST d.a1 AS u;
```
Maybe consider more descriptive alias names for clarity?

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/unnestposfun.adoc:
```diff
@@ -0,0 +1,61 @@
+
+You can use the `UNNEST_POS()` function with the xref:n1ql-language-reference/unnest.adoc[UNNEST Clause] to return the position of each element in an unnested array. 
+
+[[fn-function-name,UNNEST_POS()]]
```
needs a unique anchor. Doesn't really matter here but it will when it gets moved to `metafun.adoc`.

```suggestion
[[fn-unnest-pos,UNNEST_POS()]]
```

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/unnestposfun.adoc:
```diff
@@ -0,0 +1,61 @@
+[[fn-function-name,UNNEST_POS()]]
+== UNNEST_POS(`expr`)
+
+// Optional: only for functions with synonyms
```
Feel free to delete the template comments.

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/unnestposfun.adoc:
```diff
@@ -0,0 +1,61 @@
+== UNNEST_POS(`expr`)
+
+// Optional: only for functions with synonyms
+This function has a synonym <<fn-synonym-name>>.
```
See above re: unique anchors.

```suggestion
This function has a synonym <<fn-unnest-position>>.
```

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/unnestposfun.adoc:
```diff
@@ -0,0 +1,61 @@
+
+=== Return Values
+
+The `UNNEST_POS` function returns the position of each element in the unnested array, `expr`, as an integer. It returns each position value as a separate row in JSON format. 
```
Counting from 0. I know it's obvious from the example but I like to state it

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/unnestposfun.adoc:
```diff
@@ -0,0 +1,61 @@
+----
+====
+
+[[fn-synonym-name,UNNEST_POSITION()]]
+== UNNEST_POSITION(`expr`)
+
+Synonym of <<fn-function-name>>.
```
See above re: unique anchors

```suggestion
[[fn-unnest-position,UNNEST_POSITION()]]
== UNNEST_POSITION(`expr`)

Synonym of <<fn-unnest-pos>>.
```

simon-dew reviewed (changes requested):
Fantastic, thanks. Sorry for sparse comments, ping me if any further questions

sarahlwelton committed:
[DOC-10324] Adding notes from peer review.

Specifically, adding unique anchor link names and explicitly mentioning...

...that unnestpos counts from 0.

sarahlwelton committed:
[DOC-10324] Added UNNEST_POS to Misc Utility functions

sarahlwelton committed:
[DOC-10324] Removed UNNEST_POS from nav.adoc

simon-dew reviewed (approved):
LGTM 👍
